Dockerfile.fakemachine-arch

```diff
@@ -1,11 +1,44 @@
+# Build unpackaged dependencies from source
+FROM archlinux:base-devel AS builder
+
+# Build user-mode linux from AUR
+ENV EUID=1
```

Comment:
Haven't seen this approach before. Typically people will create separate user and su for the `makepkg`. Either approach is fine IMHO, but please add 1-2 lines worth of comments explaining why

Reply:
For what it's worth, it was a suggestion from https://www.reddit.com/r/archlinux/comments/evarc8/how_to_run_makepkg_in_docker_container_yes_as/

I've added that link and the lore around it to the Dockerfile.